CMakeLists.txt

```
@@ -0,0 +1,376 @@
+        ${FEATURE_SOURCEFILES}
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_proxy.c
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_proxy_core.c
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbhttp_digest.c
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbntlm_core.c
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbntlm_packer_std.c)
+endif()
+
+if(${USE_GZIP_COMPRESSION})
+    set(FEATURE_SOURCEFILES
+        ${FEATURE_SOURCEFILES}
+        ${CMAKE_CURRENT_LIST_DIR}/lib/miniz/miniz_tdef.c
+        ${CMAKE_CURRENT_LIST_DIR}/lib/miniz/miniz.c
+        ${CMAKE_CURRENT_LIST_DIR}/lib/pbcrc32.c
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbgzip_compress.c)
+endif()
+
+if(${RECEIVE_GZIP_RESPONSE})
+    set(FEATURE_SOURCEFILES
+        ${FEATURE_SOURCEFILES}
+        ${CMAKE_CURRENT_LIST_DIR}/lib/miniz/miniz_tinfl.c
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbgzip_decompress.c)
+endif()
+
+if(${USE_SUBSCRIBE_V2})
+    set(FEATURE_SOURCEFILES
+        ${FEATURE_SOURCEFILES}
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbcc_subscribe_v2.c
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_subscribe_v2.c)
+endif()
+
+if(${USE_ADVANCED_HISTORY})
+    set(FEATURE_SOURCEFILES
+        ${FEATURE_SOURCEFILES}
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbcc_advanced_history.c
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_advanced_history.c)
+endif()
+
+if(${USE_OBJECTS_API})
+    set(FEATURE_SOURCEFILES 
+        ${FEATURE_SOURCEFILES} 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbcc_objects_api.c 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_objects_api.c)
+endif()
+
+if(${USE_ACTIONS_API})
+    set(FEATURE_SOURCEFILES 
+        ${FEATURE_SOURCEFILES} 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbcc_actions_api.c 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_actions_api.c)
+endif()
+
+if(${USE_AUTO_HEARTBEAT})
+    set(FEATURE_SOURCEFILES 
+        ${FEATURE_SOURCEFILES} 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbauto_heartbeat.c 
+        ${CMAKE_CURRENT_LIST_DIR}/lib/pbstr_remove_from_list.c)
+    if(NOT ${OPENSSL})
+        set(FEATURE_SOURCEFILES 
+            ${FEATURE_SOURCEFILES} 
+            ${CMAKE_CURRENT_LIST_DIR}/core/pbauto_heartbeat_init_posix.c)
+    endif()
+endif()
+
+if(${USE_FETCH_HISTORY})
+    set(FEATURE_SOURCEFILES 
+        ${FEATURE_SOURCEFILES} 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_fetch_history.c 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbcc_fetch_history.c)
+endif()
+
+if(${USE_GRANT_TOKEN_API})
+    set(FEATURE_SOURCEFILES 
+        ${FEATURE_SOURCEFILES} 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbcc_grant_token_api.c 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_grant_token_api.c)
+endif()
+
+if(${USE_REVOKE_TOKEN_API})
+    set(FEATURE_SOURCEFILES 
+        ${FEATURE_SOURCEFILES} 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pbcc_revoke_token_api.c 
+        ${CMAKE_CURRENT_LIST_DIR}/core/pubnub_revoke_token_api.c)
+endif()
+
+set(OPENSSL_LINK_LIBRARIES)
+if(${OPENSSL})
+    message(STATUS "Using OpenSSL")
+
+    if(${OPENSSL_ROOT_DIR} STREQUAL "")
```

```suggestion
    if("${OPENSSL_ROOT_DIR}" STREQUAL "")
```

This line is failing locally for me. I've seen that its set at the top of the file but locally on my system this isnt initialised in time. This might be due to how CMakes caching works. The above suggestion works as expected

---

Compiles out of the box on Ubuntu 22.04! Nice work

---

Thank you for your efforts!

---

Can confirm it also works out of box using CPM which is a wrapper of CMake ExternalProject without issue and as a submodule

```
CPMAddPackage(
  NAME pubnub
  GIT_REPOSITORY https://github.com/pubnub/c-core
  GIT_TAG feat/cmake-build-system
)
```